variables.tf

```diff
@@ -469,3 +469,251 @@ variable "graphdb_enable_userdata_scripts_on_reboot" {
+  type        = bool
+  default     = true
+}
+
+variable "graphdb_parameter_store_cmk_alias" {
+  description = "The alias for the CMK key."
+  type        = string
+  default     = "graphdb-cmk-key"
+}
+
+variable "graphdb_parameter_store_key_enabled" {
+  description = "Specifies whether the key is enabled."
+  type        = bool
+  default     = true
+}
+
+variable "graphdb_parameter_store_key_spec" {
+  description = "Specification of the Key."
+  type        = string
+  default     = "SYMMETRIC_DEFAULT"
+}
+
+variable "graphdb_parameter_store_key_deletion_window_in_days" {
+  description = "The waiting period, specified in number of days for AWS to delete the KMS key(Between 7 and 30)."
+  type        = number
+  default     = 30
+}
+
+variable "graphdb_parameter_store_cmk_description" {
+  description = "Description for the KMS Key"
+  type        = string
+  default     = "KMS key for Parameter Store bucket encryption."
+}
+
+variable "enable_graphdb_parameter_store_kms_key" {
+  description = "Enable creation of KMS key for Parameter Store encryption"
+  type        = bool
+  default     = false
+}
+
+variable "parameter_store_external_kms_key" {
+  description = "Externally provided KMS CMK"
+  type        = string
+  default     = ""
+}
+
+# GraphDB EBS KMS encryption
+
+variable "graphdb_ebs_key_admin_arn" {
+  description = "ARN of the key administrator role for Parameter Store"
+  type        = string
+  default     = ""
+}
+
+variable "graphdb_ebs_key_tags" {
+  description = "A map of tags to assign to the resources."
+  type        = map(string)
+  default     = {}
+}
+
+variable "graphdb_ebs_key_rotation_enabled" {
+  description = "Specifies whether key rotation is enabled."
+  type        = bool
+  default     = true
+}
+
+variable "graphdb_ebs_cmk_alias" {
+  description = "The alias for the CMK key."
+  type        = string
+  default     = "graphdb-cmk-key"
+}
+
+variable "graphdb_ebs_key_enabled" {
+  description = "Specifies whether the key is enabled."
+  type        = bool
+  default     = true
+}
+
+variable "graphdb_ebs_key_spec" {
+  description = "Specification of the Key."
+  type        = string
+  default     = "SYMMETRIC_DEFAULT"
+}
+
+variable "graphdb_ebs_key_deletion_window_in_days" {
+  description = "The waiting period, specified in number of days for AWS to delete the KMS key(Between 7 and 30)."
+  type        = number
+  default     = 30
+}
+
+variable "graphdb_ebs_cmk_description" {
+  description = "Description for the KMS Key"
+  type        = string
+  default     = "KMS key for Parameter Store bucket encryption."
+}
+
+variable "enable_graphdb_ebs_kms_key" {
+  description = "Enable creation of KMS key for Parameter Store encryption"
+  type        = bool
+  default     = false
+}
+
+variable "ebs_external_kms_key" {
+  description = "Externally provided KMS CMK"
+  type        = string
+  default     = ""
+}
+
+variable "ebs_cmk_enabled" {
```

Comment:
rename it to ebs_encrypt_with_cmk_key for example.

Reply:
Renamed to `create_graphdb_ebs_kms_key` for more clarity